inductiva/tests/mixins/test_file_manager.py

```diff
@@ -0,0 +1,201 @@
+def test_add_files__valid_file__render_totarget(source_file, render_args,
+                                                target_file):
+    """Check that a non-empty input file is render to the given target.
+    
+    Goal: Test another functionality of the FileManager().add_file method, which
+    renders template files from render args into a new target file. The render
+    process automatically copies and renders the file on the spot. In case,
+    target_file is None, then the target_file name will be the source file
+    without the template extension.
+    """
+
+    file_manager = mixins.FileManager()
+    target_file = file_manager.add_file(source_file, target_file, **render_args)
+    root_dir = file_manager.get_root_dir()
+    assert os.path.isfile(os.path.join(root_dir, target_file))
+
+
+@pytest.mark.parametrize("source_dir, target_dir", [
+    (TEST_DIR, None),
+    (TEST_DIR, "added_dir"),
+])
+def test_add_dir__valid_dir__no_args(source_dir, target_dir):
+    """Check that an existing input dir is added to the given target dir.
+    
+    Goal: Test one of the functionalities of FileManager().add_dir which just
+    copies a source dir into a new target dir. If no arguments are passed, then
+    no rendering occurs, which means that even template files are copied as is.
+    All files are copied with the same exact name.
+    """
+
+    file_manager = mixins.FileManager()
+    target_dir = file_manager.add_dir(source_dir, target_dir)
+
+    root_dir = file_manager.get_root_dir()
+    target_path = os.path.join(root_dir, target_dir)
+    assert os.path.isdir(target_path)
+    assert os.listdir(TEST_DIR) == os.listdir(target_path)
+
+
+@pytest.mark.parametrize("source_dir, render_args, target_dir", [(TEST_DIR, {
+    "text": "yes",
+    "simulator": "openfoam"
+}, None), (TEST_DIR, {
+    "text": "yes",
+    "simulator": "openfoam"
+}, "rendered_dir")])
+def test_add_dir__valid_dir__render_args(source_dir, render_args, target_dir):
+    """Check that an input dir with template files is render to a target dir.
+    
+    Goal: Test another of the functionalities of FileManager().add_dir which
+    copies a directory into another one and renders the template files from the
+    given render_args. All files are copied, even the ones that aren't template
+    files. Then, the template files are rendered and keep the same name but
+    without the template extension.
+    Hence, the rendering occurs only over files with the template extension and
+    if render_args are passed. Further, if render_args doesn't contain all
+    arguments that are on the template files, then those extra arguments are 
+    rendered with None. We leave this to the responsability of the user.
+    """
+    file_manager = mixins.FileManager()
+    target_dir = file_manager.add_dir(source_dir, target_dir, **render_args)
+
+    root_dir = file_manager.get_root_dir()
+    target_path = os.path.join(root_dir, target_dir)
+    assert os.path.isdir(target_path)
```

Comment:
this is testing `get_root_dir` but the test is testing `add_dir`.
This assertion should be moved to it's own test.